SIGSEGV in `Dialer.GetDNSDataFromCache`

**Steps to Reproduce:**

* This happens when an interruption occurs in Nuclei.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xcea227]

goroutine 58375 [running]:
github.com/projectdiscovery/fastdialer/fastdialer.(*Dialer).GetDNSDataFromCache(0x0, {0xc0182118c8?, 0xc0182118c8?})
	/home/dw1/.go/pkg/mod/github.com/projectdiscovery/fastdialer@v0.2.7/fastdialer/dialer.go:345 +0x47
github.com/projectdiscovery/fastdialer/fastdialer.(*Dialer).GetDNSData(0x0, {0xc0182118c8?, 0x0?})
	/home/dw1/.go/pkg/mod/github.com/projectdiscovery/fastdialer@v0.2.7/fastdialer/dialer.go:390 +0x33c
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).addCNameIfAvailable(0x27d1ca0?, {0xc0182118c8?, 0x2bff6eb?}, 0xc01a0820f0)
	/home/dw1/Development/PD/nuclei/pkg/protocols/http/request.go:1045 +0x28
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).executeRequest(0xc0084663c0, 0xc01604b200, 0xc007cf31a0, 0xc0064bff58, 0x0, 0xc018266be0?, 0x0)
	/home/dw1/Development/PD/nuclei/pkg/protocols/http/request.go:951 +0x289b
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).executeParallelHTTP.func3(0xc007cf31a0)
	/home/dw1/Development/PD/nuclei/pkg/protocols/http/request.go:308 +0x2d8
created by github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).executeParallelHTTP in goroutine 54484
	/home/dw1/Development/PD/nuclei/pkg/protocols/http/request.go:292 +0x37c
```

--- a/pkg/protocols/http/request.go
+++ b/pkg/protocols/http/request.go
@@ -1042,6 +1042,10 @@ func (request *Request) validateNFixEvent(input *contextargs.Context, gr *genera
 
 // addCNameIfAvailable adds the cname to the event if available
 func (request *Request) addCNameIfAvailable(hostname string, outputEvent map[string]interface{}) {
+	if protocolstate.Dialer == nil {
+		return
+	}
+
 	data, err := protocolstate.Dialer.GetDNSData(hostname)
 	if err == nil {
 		switch len(data.CNAME) {

